smithy-model/src/main/java/software/amazon/smithy/model/loader/AstModelLoader.java

```diff
@@ -293,6 +293,15 @@ private void loadResource(ShapeId id, ObjectNode node, FullyResolvedModelFile mo
             }
         });
 
+        // Load properties and resolve forward references.
+        node.getObjectMember("properties").ifPresent(properties -> {
+            for (Map.Entry<StringNode, Node> entry : properties.getMembers().entrySet()) {
+                String name = entry.getKey().getValue();
+                ShapeId target = loadReferenceBody(modelFile, id, entry.getValue());
+                builder.addProperty(name, target);
```

Comment:
nit: we usually use "put" for map semantics

Reply:
I was keeping builder method name consistent with `addIdentifier()` but the underlying call is "put."